src/metabase/driver/generic_sql.clj

```diff
@@ -325,20 +325,29 @@
          {:name   (:table_name table)
           :schema (:table_schem table)})))
 
-(defn- describe-table-fields
-  [^DatabaseMetaData metadata, driver, {:keys [schema name]}]
-  (set (for [{:keys [column_name type_name]} (jdbc/result-set-seq (.getColumns metadata nil schema name nil))
-             :let [calculated-special-type (column->special-type driver column_name (keyword type_name))]]
-         (merge {:name      column_name
-                 :custom    {:column-type type_name}
-                 :base-type (or (column->base-type driver (keyword type_name))
-                                (do (log/warn (format "Don't know how to map column type '%s' to a Field base_type, falling back to :type/*."
-                                                      type_name))
-                                    :type/*))}
-                (when calculated-special-type
-                  (assert (isa? calculated-special-type :type/*)
-                    (str "Invalid type: " calculated-special-type))
-                  {:special-type calculated-special-type})))))
+(defn- database-type->base-type
+  "Given a `database-type` (e.g. `VARCHAR`) return the mapped Metabase type (e.g. `:type/Text`)."
+  [driver database-type]
+  (or (column->base-type driver (keyword database-type))
+      (do (log/warn (format "Don't know how to map column type '%s' to a Field base_type, falling back to :type/*."
+                            database-type))
+          :type/*)))
+
+(defn- calculated-special-type
+  "Get an appropriate special type for a column with `column-name` of type `database-type`."
+  [driver column-name database-type]
+  (when-let [special-type (column->special-type driver column-name (keyword database-type))]
+    (assert (isa? special-type :type/*)
+      (str "Invalid type: " special-type))
+    special-type))
+
+(defn- describe-table-fields [^DatabaseMetaData metadata, driver, {schema :schema, table-name :name}]
+  (set (for [{database-type :type_name, column-name :column_name} (jdbc/result-set-seq (.getColumns metadata nil schema table-name nil))]
+         (merge {:name          column-name
+                 :database-type (name database-type)
```

Comment:
This will always be a string out of the database, right?

Reply:
yeah. It will. good point